### PR TITLE
[CDAP-9532] Adds Encode and Decode point and click directives to dataprep

### PR DESCRIPTION
**Pull request by ajainarayanan**
- Adds EncodeDecode component to handle both encoding and decoding data in dataprep 
- Disables certain decode options that require the column to be of type `byte[]`.

JIRA: https://issues.cask.co/browse/CDAP-9532

### CODE REVIEW

**Comment by tonybach** [comment]
screen shot 2017-06-13 at 4 01 09 pm

Not exactly related to this PR, but this is what a maximum-height window looks like on my 13-inch Macbook without going full screen. I can't even see Encode or Decode at all. Not sure how to handle this though, since it's inevitable that this will happen as we add more directives :( 

**EDIT**: Oh just noticed there's https://issues.cask.co/browse/CDAP-11886.

File: cdap-ui/app/cdap/components/DataPrep/Directives/EncodeDecode/EncodeDecode.scss [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+      cursor: pointer;
+      line-height: 1.5;
+      font-size: 13px;
+
+      &.disabled {
+        cursor: not-allowed;
+        pointer-events: auto;
+      }
+      &:hover {
+        background-color: #dddddd;
```
Comment (tonybach) [inline cdap-ui/app/cdap/components/DataPrep/Directives/EncodeDecode/EncodeDecode.scss]: Extract this color?

**Comment by tonybach** [comment]
One minor comment, otherwise code LGTM 👍

**Comment by ajainarayanan** [comment]
cc: @elfenheart I have modified calculating second level popover offset to align with the main menu item. Let me know if I missed a scenario because I have removed the footer height while calculating the body bottom.

**Comment by elfenheart** [comment]
@ajainarayanan do we want it to overflow over the footer?

**Review by elfenheart — APPROVED** [review]
Just a question, otherwise 👍  LGTM

**Comment by ajainarayanan** [comment]
@elfenheart Have modified it so that the second level popover doesn't hover on top of footer. @tonybach Have fixed the weird second level popover issue you are seeing. Reg the popover going beyond the viewport have another PR that addresses that: https://github.com/caskdata/cdap/pull/9109

**Comment by bdmogal** [comment]
@tonybach let's add a horizontal rule between the various decoding params and character sets?
With a heading "Set Charset"

**Comment by tonybach** [comment]
Discussed with @bdmogal, will move `UTF-8`, `UTF-16`, `US-ASCII` etc. from the `Decode` menu and move it to a new one called `Set Character Encoding`.

File: cdap-ui/app/cdap/components/DataPrep/Directives/EncodeDecode/index.js [hunk 2]
```diff
@@ -117,20 +113,31 @@ export default class EncodeDecode extends Component {
...
+          <span className="float-xs-right">
+            <span className="fa fa-caret-right" />
+          </span>
 
-        {this.renderDetail()}
+          {this.renderDetail()}
+        </div>
+        {
+          this.props.isDisabled && this.props.disabledTooltip ? (
+            <UncontrolledTooltip target={id} delay={{show: 250, hide: 0}}>
```
Comment (elfenheart) [inline cdap-ui/app/cdap/components/DataPrep/Directives/EncodeDecode/index.js]: new line for each of the attributes

**Review by elfenheart — APPROVED** [review]
👍  LGTM

**Comment by ajainarayanan** [comment]
Have a LGTM from @bdmogal . Merging.

**Pull request merged**